plugins/azure/fix_plugin_azure/resource/compute.py

```diff
@@ -3286,6 +3292,26 @@ class AzureVirtualMachineScaleSet(AzureResource, BaseAutoScalingGroup):
     upgrade_policy: Optional[AzureUpgradePolicy] = field(default=None, metadata={'description': 'Describes an upgrade policy - automatic, manual, or rolling.'})  # fmt: skip
     virtual_machine_profile: Optional[AzureVirtualMachineScaleSetVMProfile] = field(default=None, metadata={'description': 'Describes a virtual machine scale set virtual machine profile.'})  # fmt: skip
     zone_balance: Optional[bool] = field(default=None, metadata={'description': 'Whether to force strictly even virtual machine distribution cross x-zones in case there is zone outage. Zonebalance property can only be set if the zones property of the scale set contains more than one zone. If there are no zones or only one zone specified, then zonebalance property should not be set.'})  # fmt: skip
+    _vmss_vm_ids: Optional[List[str]] = field(default=None)
+
+    def post_process(self, graph_builder: GraphBuilder, source: Json) -> None:
+        def collect_vmss_instances() -> None:
+            api_spec = AzureApiSpec(
+                service="compute",
+                version="2023-09-01",
+                path=f"{self.id}/virtualMachines",
+                path_parameters=[],
+                query_parameters=["api-version"],
+                access_path="value",
+                expect_array=True,
+            )
+
+            items = graph_builder.client.list(api_spec)
+            self._vmss_vm_ids = [str(item.get("id")) for item in items if item.get("id") is not None]
```

Comment:
Instead of storing the ids in a property to connect the instances, you could add the edge here.